### DISCUSSION TITLE
GraphQL pagination response, probably displays wrong results in V4

### DISCUSSION THREAD

**Discussion by MatteoGioioso** [community/community#24566] [Apps, API and Webhooks]
<p>I think the response I got from my paginated query might be wrong.<br>
This is my query:</p>
<pre><code>query($language: String!, $endCursor: String, $numberOfRepos: Int){
  search(query: $language, type: REPOSITORY, first: $numberOfRepos, after: $endCursor) {
    pageInfo {
      startCursor
      hasNextPage
      endCursor
    }
    edges {
      node {
        ... on Repository {
          nameWithOwner
          isArchived
          stargazers{
              totalCount
          }
        }
      }
    }
  }
}
</code></pre>
<p>And those are my input variables</p>
<pre><code>{
	"language": "stars:&gt;=500 sort:stars",
	"endCursor": "Y3Vyc29yOjk5OQ==",
	"numberOfRepos": 1
}
</code></pre>
<p>And finally this is the response I got:</p>
<pre><code>{
    "data": {
        "search": {
            "pageInfo": {
                "startCursor": "Y3Vyc29yOjEwMDA=",
                "hasNextPage": false,
                "endCursor": "Y3Vyc29yOjEwMDA="
            },
            "edges": [
                {
                    "node": {
                        "nameWithOwner": "foreversd/forever",
                        "isArchived": false,
                        "stargazers": {
                            "totalCount": 12813
                        }
                    }
                }
            ]
        }
    }
}
</code></pre>
<p>Now as you can see this repo has about 12k stars and I got hasNextPage false.<br>
I believe that this is not correct, I cannot believe that there are no more repositories with more than 500 stars after this one D=.</p>
<p>Am I missing something? Is my query wrong?</p>
<p>I want to get all the repositories with stars count bigger than 500 and ordered by starts count from big to small.</p>
<p>Any idea? <br>
Thanks</p>

**Comment by guntrip** [comment, marked as answer]
<p>Hi Matteo,</p>
<p>Both the REST API and GraphQL API <a href="https://developer.github.com/v3/search/#search" rel="nofollow noopener">cap the  number of returned results at 1000</a> and the <strong>after</strong> argument in your query is pointing to the very last result that can be returned (as the API currently uses base64 to encode these values, we can confirm that “<strong>Y3Vyc29yOjEwMDA=</strong>” is " <strong>cursor:999</strong>"). Because there’s no more results we can return,   <strong>hasNextPage: false</strong> is expected here.</p>
<p>If you wanted to work around the 1000 result limit, I’d suggest trying to further narrow your search. You can use ranges (such as <strong>stars:500…600</strong>) but as this is searching all repositories on GitHub, you might find you need to filter those results further. Here’s an example that also filters by language:</p>
<pre><code>query {
  search(query: "stars:500..600 language:ruby sort:stars", type: REPOSITORY, first: 20) {
    pageInfo {
      startCursor
      hasNextPage
      endCursor
    }
    repositoryCount
    edges {
      node {
        ... on Repository {
          nameWithOwner
          isArchived
          stargazers{
              totalCount
          }
        }
      }
    }
  }
}
</code></pre>
<p>Note that I’ve included <strong>repositoryCount</strong> too which allows us to confirm the total number of results we’re working with.</p>
<p>I hope this helps!</p>
